processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -79,13 +86,39 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    throw new UnsupportedOperationException();
+    return factory.makeValueMatcher(dimension, new Predicate<String>()
+    {
+      @Override public boolean apply(String input)
+      {
+        // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+        return value.equals(fn.apply(Strings.emptyToNull(input)));
+      }
+    });
   }
 
   @Override
-  public ValueMatcher makeMatcher(ColumnSelectorFactory factory)
+  public ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory)
   {
-    throw new UnsupportedOperationException();
+    final DimensionSelector dimensionSelector = columnSelectorFactory.makeDimensionSelector(dimension, null);
+    if (dimensionSelector == null) {
+      return new BooleanValueMatcher(Strings.isNullOrEmpty(fn.apply(value)));
```

Comment:
i am not sure when `dimensionSelector == null` but following the same logic in other filter it should be  `return new BooleanValueMatcher(value.equals(fn.apply(null)))` i guess.